server/grpc/server.go

```diff
@@ -82,3 +87,19 @@ func (s *Server) register() {
 
 	reflection.Register(s.instance)
 }
+
+func streamCosmosErrorInterceptor(srv interface{}, stream grpc.ServerStream, info *grpc.StreamServerInfo, handler grpc.StreamHandler) error {
```

Comment:
what about a new package `error` containing a struct `error` that both implement the cosmos error and the grpc status code (`GRPCStatus() *status.Status`)?

Reply:
But why you want to implement that, if you already have the function for doing that.
First of all, we don't have a type for errors in `cosmos/errors.go`, we just retruns `cosmostypes.Error`, but even if there will be type for it, the GRPCStatus interface will looks like this

```

type Error types.ErrorImpl

func (e *Error) GRPCStatus() * status.Status {
   return status.Error(codes.Code(e.Code()), e.Error())
}
```

So exacly the same as interceptor.

So to sumamriazie, we can't and we shound't implement that interface.